### DISCUSSION TITLE
Which Windows versions are/will be supported?

### DISCUSSION THREAD

**Discussion by Greedquest** [twinbasic/twinbasic#715] [Q&A]
Aside from potential plans to leave vscode in favour of a dedicated IDE, what are the current versions of Windows which:
- twinBASIC can be programmed in (i.e. which versions of vscode are supported and what platforms do they run on)
- twinBASIC code can target once compiled (I'm hoping the same platforms as VBx already, I know more will come on the horizon - although is there an early version/platform that won't be supported - e.g. win 16, idk if vb6 could target this)

---
**For background:**
I'm making my own tool and I'm wondering what's the earliest version of Windows I realistically need to support. So if anyone else has thoughts on that I'd be keen to hear (although only tangentially related to tB). It's a tradeoff between dropping Win 7 or not, XP and earlier would be a challenge to support.

**Comment by bclothier** [comment]
I will leave it up to Wayne to comment on how widely tB can run.

However for _building_ tB projects, I imagine we are bound to [the OS requirements of VSC](https://code.visualstudio.com/Docs/supporting/requirements).

**Comment by Kr00l** [comment]
We need to split between IDE requirements and compiled "runtime" requirements.

As bclothier pointed out that the IDE requirement is a constant dictated by vs code, which seems to be Windows 7.

For runtime support I can't say and I am curios what Wayne comments about this.
But I already tested an tB exe file and it runs on WinXP, even themed.

**Reply by mburns08109** [reply]
Oohh! Oooh! Say Win95, Wayne!  :-)

**Reply by Kr00l** [reply]
I think it can't be lower as Win2k due to unicode API usage in tB forms for example.
But I think it's just luck or not if it works or not.

**Reply by WaynePhillipsEA** [reply]
Indeed, the official line will be that twinBASIC generated executables will support XP / Win2k onwards, initially.

Once all the hard work is done to support GUI, we might **_consider_** offering an ANSI package of all the GUI stuff (for the ultimate backwards-compatibility)... but that really depends on demand.   I'm not sure in 2022 whether it's worth the effort to offer ANSI builds anymore.

**Reply by Kr00l** [reply]
IMHO since the main message bump is unicode in tB. (which is a huge win) it does not make sense to partially offer ANSI components.
Not offering ANSI components doesn't mean old codebases which use SendMessageA for example won't work anymore. They continue to work for unicode windows as well.
So I can't see a need here, if I'm not mistaken.

**Reply by WaynePhillipsEA** [reply]
The message pump could be adjusted for an ANSI build quite easily.   But like you say... it's whether there is a need for it.   I'm sure there'll be a few bugs in migrating some existing apps caused by the change to unicode... but time will tell whether we actually need to provide full ANSI builds.

**Reply by wqweto** [reply]
IMO thete is very slim chance 9x support would benefit anyone. Even for embedded a slim Linux distro would be more appropriate. 

I would rather embrace Unicode support in the very core incl. long filenames 100% support (or at least best effort with shortnames shenanigans etc.)

**Reply by WaynePhillipsEA** [reply]
Cool.   Long filename support should all be done, though it is one of the less tested areas at the moment.
